### ISSUE TITLE
Error on data source does not show additional info with line and resource where it happened

### ISSUE THREAD

**Issue by marekhanzlikz** [hashicorp/terraform-provider-aws#15301]
<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform AWS Provider Version
Terraform v0.12.29
Terraform v0.13.3

provider aws v3.7.0
<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

data sources

### Expected Behavior
```
Error: no matching VPC found                                                                                                                                                                                       
                                                                                                                                                                                                                   
  on data.tf line 51, in data "aws_vpc" "test-pen":                                                                                                                                                                
  51: data "aws_vpc" "test-pen" {
```
### Actual Behavior

```
Error: no matching VPC found
```
### Steps to Reproduce

Create a data source that cant possibly return a value and you get an error that the resource was not found but it does not tell you what resource it is and on what line

With terraform 12.28 and AWS provider v2.63.0 all worked as expected but after update to latest (and tried terraform 0.13 too) it does not show helping information

**Comment by bflad** [comment]
This occurred after upgrading to Terraform Plugin SDK version 2 -- upstream bug report: https://github.com/hashicorp/terraform-plugin-sdk/issues/561

**Comment by aRobinson-R7** [comment]
Due to the same issue, this makes some more generic error messages very hard to debug (searching for an AMI)
```
Error: Your query returned no results. Please change your search criteria and try again
```

**Comment by bflad** [comment]
The fix for this (an upgrade to our Terraform Plugin SDK dependency) has been merged and will release with version 3.30.0 of the Terraform AWS Provider, later this week. Please note that using Terraform CLI 0.15.0-beta1 (and later) also contains a fix that can also remediate the issue without the provider upgrade.
